docs/DeveloperGuide.md

```diff
@@ -4,26 +4,207 @@
+
+Given below is the steps on how the `add_meeting` operation works.
+
+>To be added
+> 
+## 3.2 Listing Events Feature
+The `list` command is a command that the user can input in order to list out the events he has in his timetable.
+
+`list [user]` displays the timetable for the particular user.
+
+`list all` displays the timetable for all users.
+
+In essence, the `list all` command repeatedly calls `list [user]` for all users.
+
+Before the timetable is listed out, it will also be sorted according to day and time for easy reading.
+
+The following sequence diagram shows how the command `` is executed.
+![ListCommandSequenceDiagram](diagrams/ListCommandSequenceDiagram.png)
+
+## 3.3 Finding Common Free Timeslots Feature
+The `free` command is a command that the user can input in order to find timeslots where all users are free.
+
+For greater customisation, `free [duration]` displays all common timeslots which has a duration longer than specified.
+
+The searching algorithm works by marking the busy slots of each timetable first.
+Timings that are not marked 'busy' are then identified as free time slots.
+
+The following sequence diagram shows how the command `free` is executed.
+![FreeCommandSequenceDiagram](diagrams/FreeCommandSequenceDiagram.png)
+
+## 3.4 Delete events `delete`
+Deletes an event from the user's specified timetable
+
+**Format:** ` delete n/NAME i/INDEX`
+
+* Deletes from the timetable of user
+* Deletes the lesson at the specified `INDEX`.
+* The `INDEX` refers to the index number shown in the displayed lesson list.
+
+Example of usage:
+
+`delete n/John i/1`
+
+If the event is a meeting, it will delete this event from all users
+
+The following sequence diagram shows how the `delete` command works:
+![DeleteCommandSequenceDiagram](diagrams/DeleteCommandSequenceDiagram.png)
+
+Before the timetable is listed out, it will also be sorted according to day and time for easy reading.
+
+## 3.4 Data Saving Feature
+
+The save mechanism is facilitated by StorageFile. It creates a text file in the local machine and writes all the elements
+in the MasterTimetable and Timetable to the text file in human-readable format.
+
+Given below is the steps on how the save operation works.
+
+Step 1: When the user exits the application. The StorageFile.saveData will be initialized.
+
+Step 2: It will check for the existence of the data file. If the file is not found, it will create the file.
+Else do nothing.
+
+Step 3: After creating the file, collateAll method will be called to consolidate all elements in the MasterTimetable and Timetable list,
+and write them into the data file.
+
+Step 4: A message will be shown to the user, informing him that the data has been saved successfully. For any exception
+encountered along the way, an error message will be shown.
+
+The following sequence diagram shows how the save operation works:
+![SaveSequenceDiagram](diagrams/SaveSequenceDiagram.png)
```

Comment:
Should the lifeline extend past the point where an object is deleted?
![image](https://user-images.githubusercontent.com/26570382/161183631-94225ec6-ba5b-4a81-9f31-5ec1dba2bce6.png)